Factory/BeverageFactory.js

```diff
@@ -0,0 +1,20 @@
+import Coffee from './../Coffee.js';
+import Tea from './../Tea.js';
+
+const classes = {
+    Coffee: Coffee,
+    Tea,
+};
+
+export default class BeverageFactory {
+
+    static MakeBeverage(type, ...params) {
+        return new classes[type](...params);
+    }
+
```

Comment:
Error handling would be nice here:
 
 if (!classes[type]) {
            throw new Error(`Beverage type '${type}' not supported.`);
        }